### PR TITLE
CLDR-13576 Add Ligurian core data

### PR DESCRIPTION
**Pull request by jeanm**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-13576
- [x] Updated PR title and link in previous line to include Issue number

##### Comments

- Ran tests as decribed in <http://cldr.unicode.org/development/running-tests>:
  - `ant datacheck` passes.
  - `ant unittestExhaustive` has no new failures compared to base revision. Detailed results [here](https://gist.github.com/jeanm/1853e8beff6a6646f55eb239fb34b689).

### CODE REVIEW

**Comment by jeanm** [comment]
Hi, any chance I could get a review of this and the related ticket on JIRA? Let me know if anything needs fixing :)

**Review by srl295 — DISMISSED** [review]
looks reasonable

**Comment by srl295** [comment]
@jeanm if you rebase this on master, the tests may pass. Make sure its one commit.

test failures seem unrelated.

**Comment by srl295** [comment]
@jeanm also because this is new, please move the files from:

common/main/lij* into seed/main/
common/casing/lij* into seed/casing/
common/collation/lij* into seed/collation/

thanks

**Comment by srl295** [comment]
@jeanm this was approved for 38 if we can get the above changes in.

**Comment by srl295** [comment]
@jeanm this was approved for 38 if we can get the above changes in.

**Comment by srl295** [comment]
(Sorry did not mean to close)

**Comment by jeanm** [comment]
Thanks for the review @srl295! I am now getting failures for `CLDR/TestCLDRFile/testExtraPaths` and `CLDR/TestCLDRLocaleCoverage/TestLanguageNameCoverage`. Could the first failure be related to having moved files around? I'm not sure the second failure is related to my changes.

File: common/supplemental/coverageLevels.xml [hunk 1]
```diff
@@ -99,7 +99,7 @@ For terms of use, see http://www.unicode.org/copyright.html
...
-		<coverageVariable key="%language80" value="(doi|a([fkmrz]|gq|s[at]?)|b([gm-os]|as|e[mz]?|rx?)|c([aosy]|cp|eb?|gg|hr|kb)|d([ez]|av?|je|sb|ua|yo)|e([elnos-u]|bu|wo)|f([afory]|il?|ur)|fa_AF|g([adlv]|sw|uz?)|h([eirtuy]|aw?|mn|sb)|i[adgist]|j([av]|go|mc)|k([imnuwy]|a[bm]?|de|ea|hq|kj?|ln?|ok?|s[bfh]?)|l([bgnotv]|ag?|kt|rc|u[oy]?)|m([iklr-ty]|a[is]|er|fe|g[ho]?|ni?|u[al]|zn)|n([bel]|aq|ds?|mg|nh?|us|yn?)|o[mrs]|p([alst]|cm)|qu|r([mnu]|of?|wk?)|s([dgiklnoqrt-w]|a[hqt]?|bp|e[hs]?|hi|mn?)|t([ag-ikort]|eo?|wq|zm)|u([gkrz]|nd)|v([i]|ai|un)|w(ae|o)|x(h|og)|y([io]|av|ue)|z([hu]|gh|xx))"/>
+		<coverageVariable key="%language80" value="(doi|a([fkmrz]|gq|s[at]?)|b([gm-os]|as|e[mz]?|rx?)|c([aosy]|cp|eb?|gg|hr|kb)|d([ez]|av?|je|sb|ua|yo)|e([elnos-u]|bu|wo)|f([afory]|il?|ur)|fa_AF|g([adlv]|sw|uz?)|h([eirtuy]|aw?|mn|sb)|i[adgist]|j([av]|go|mc)|k([imnuwy]|a[bm]?|de|ea|hq|kj?|ln?|ok?|s[bfh]?)|l([bgnotv]|ag?|ij|kt|rc|u[oy]?)|m([iklr-ty]|a[is]|er|fe|g[ho]?|ni?|u[al]|zn)|n([bel]|aq|ds?|mg|nh?|us|yn?)|o[mrs]|p([alst]|cm)|qu|r([mnu]|of?|wk?)|s([dgiklnoqrt-w]|a[hqt]?|bp|e[hs]?|hi|mn?)|t([ag-ikort]|eo?|wq|zm)|u([gkrz]|nd)|v([i]|ai|un)|w(ae|o)|x(h|og)|y([io]|av|ue)|z([hu]|gh|xx))"/>
```
Comment (srl295) [inline common/supplemental/coverageLevels.xml:102]:
```suggestion
		<coverageVariable key="%language80" value="(doi|a([fkmrz]|gq|s[at]?)|b([gm-os]|as|e[mz]?|rx?)|c([aosy]|cp|eb?|gg|hr|kb)|d([ez]|av?|je|sb|ua|yo)|e([elnos-u]|bu|wo)|f([afory]|il?|ur)|fa_AF|g([adlv]|sw|uz?)|h([eirtuy]|aw?|mn|sb)|i[adgist]|j([av]|go|mc)|k([imnuwy]|a[bm]?|de|ea|hq|kj?|ln?|ok?|s[bfh]?)|l([bgnotv]|ag?|kt|rc|u[oy]?)|m([iklr-ty]|a[is]|er|fe|g[ho]?|ni?|u[al]|zn)|n([bel]|aq|ds?|mg|nh?|us|yn?)|o[mrs]|p([alst]|cm)|qu|r([mnu]|of?|wk?)|s([dgiklnoqrt-w]|a[hqt]?|bp|e[hs]?|hi|mn?)|t([ag-ikort]|eo?|wq|zm)|u([gkrz]|nd)|v([i]|ai|un)|w(ae|o)|x(h|og)|y([io]|av|ue)|z([hu]|gh|xx))"/>
```

@jeanm  Missed this. Please revert this line- this says, "everyone ought to translate the term Ligurian".  Please don't mandate that yet.

**Comment by srl295** [comment]
@jeanm can you apply this change:

```diff
diff --git a/tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCLDRFile.java b/tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCLDRFile.java
index 9137bf2e88..f4a71dbbd7 100644
--- a/tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCLDRFile.java
+++ b/tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCLDRFile.java
@@ -195,7 +195,7 @@ public class TestCLDRFile extends TestFmwk {
         Counter<String> extraPaths = new Counter<>();
         for (String locale : sdi.hasGrammarInfo()) {
             if (sdi.getGrammarInfo(locale).hasInfo(GrammaticalTarget.nominal)) {
-                final CLDRFile cldrFile = CLDRConfig.getInstance().getCldrFactory().make(locale, true);
+                final CLDRFile cldrFile = CLDRConfig.getInstance().getFullCldrFactory().make(locale, true);
                 Set<String> sorted2 = new TreeSet<>(cldrFile.getExtraPaths());
                 for (String path : sorted2) {
                     if (path.contains("/gender") || path.contains("@gender") || path.contains("@case")) {
```

Actually, i'll open a PR against you for it.

**Comment by srl295** [comment]
@jeanm I opened a "sub-PR" against you https://github.com/jeanm/cldr/pull/1  — this should fix issues.
If you like it, accept and merge it, then rebase and squash again. Then we can merge.

Kindly add this line:

```
Co-authored-by: Steven R. Loomis <srl295@gmail.com>
```

**Comment by srl295** [comment]
@btangmu @macchiati @pedberg-icu PTAL at my test code fixes now in this PR. A clarification and a fix.

File: tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCLDRFile.java [hunk 2]
```diff
@@ -195,7 +195,7 @@ public void testExtraPaths() {
         Counter<String> extraPaths = new Counter<>();
         for (String locale : sdi.hasGrammarInfo()) {
             if (sdi.getGrammarInfo(locale).hasInfo(GrammaticalTarget.nominal)) {
-                final CLDRFile cldrFile = CLDRConfig.getInstance().getCldrFactory().make(locale, true);
+                final CLDRFile cldrFile = CLDRConfig.getInstance().getFullCldrFactory().make(locale, true);
```
Comment (srl295) [inline tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCLDRFile.java:198]: If there is grammar info, check it, even if it's in seed. (there's no seed/supplemental/* )

File: tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCLDRLocaleCoverage.java [hunk 3]
```diff
@@ -171,7 +171,7 @@ static final StringBuilder composeList(Iterable<String> source, String separator
     }
 
     private boolean assertContains(String title, Collection<String> set, Collection<String> subset) {
-        boolean result = assertTrue(title, set.containsAll(subset));
+        boolean result = set.containsAll(subset);
```
Comment (srl295) [inline tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCLDRLocaleCoverage.java:174]: Thought i fixed this before. Basically, this makes sure we get some details about what locales were missing, not just 'assert failed'.

The issue here was that this PR had incorrectly added `lij` to language80 coverage rules. That's no longer in this PR, but this makes sure that the resulting err is clear.

File: tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCLDRLocaleCoverage.java [hunk 4]
```diff
@@ -181,6 +181,7 @@ private boolean assertContains(String title, Collection<String> set, Collection<
             }
             warnln("Missing:\t" + temp.size() + "\n\t" + Joiner.on("\n\t").join(temp2));
         }
+        assertTrue(title, result);
```
Comment (srl295) [inline tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCLDRLocaleCoverage.java:184]: we still assert, but print the warning first before bailing out

**Comment by srl295** [comment]
@jeanm forgot one more thing ( actually, a task I'm doing at the same time in another window) https://github.com/jeanm/cldr/pull/2

**Comment by jeanm** [comment]
@srl295 Thanks for all the help! Merged, rebased, and squashed.

**Review by srl295 — APPROVED** [review]

**Review by pedberg-icu — APPROVED** [review]

**Comment by srl295** [comment]
Merging, and will watch master this time!

**Comment by srl295** [comment]
@jeanm this was merged a week ago… if you have updated data, please make a new PR (or just wait to update in the survey tool) thanks

**Pull request merged**